### PR TITLE
Bounds and project tidying

### PR DESCRIPTION
**Pull request by michaelpj**
Pulled out of https://github.com/haskell/lsp/pull/377

### CODE REVIEW

File: lsp-test/lsp-test.cabal [hunk 1]
```diff
@@ -99,7 +99,7 @@ test-suite func-test
   main-is:             FuncTest.hs
   hs-source-dirs:      func-test
   type:                exitcode-stdio-1.0
-  build-depends:       base <4.16
+  build-depends:       base
```
Comment (jneira) [inline lsp-test/lsp-test.cabal:102]: should not have the package base bounds?

File: .github/workflows/haskell.yml [hunk 2]
```diff
@@ -20,7 +20,7 @@ jobs:
     - uses: haskell/actions/setup@v1
       with:
         ghc-version: ${{ matrix.ghc }}
-        cabal-version: '3.4'
+        cabal-version: 'latest'
```
Comment (jneira) [inline .github/workflows/haskell.yml:23]: not always last cabal is good, would `3.6` be too much careful?

**Review by jneira — APPROVED** [review]
look good overall, thanks for the cleaning up!

File: .github/workflows/haskell.yml (hunk 2 above)
Comment (jneira) [reply .github/workflows/haskell.yml:23]: afaik 3.6 will pick 3.6.2.0, 3.6.0.0 suffered some bugs

Comment (michaelpj) [reply .github/workflows/haskell.yml:23]: Fair enough.

File: lsp-test/lsp-test.cabal (hunk 1 above)
Comment (michaelpj) [reply lsp-test/lsp-test.cabal:102]: It depends on `lsp-test` which has the bounds - adding them here just means extra work IMO.

Comment (jneira) [reply lsp-test/lsp-test.cabal:102]: oh I miss the existing bound in the package lib

**Pull request merged**